File: addressbook-main/src/main/java/com/addressbook/dto/PageDataDto.java
package com.addressbook.dto;

import com.addressbook.model.FieldDescription;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.Map;

@Getter
@Setter
@Builder
public class PageDataDto<T> {

    private T data;
    private Map<String, FieldDescription> fieldDescriptionMap;
}

File: addressbook-common/src/main/kotlin/com/addressbook/dto/PageDataDto.kt
package com.addressbook.dto

class PageDataDto<T>(data: T?, fieldDescriptionDtoMap: Map<String, FieldDescriptionDto>?) {

    var data: T? = null
    var fieldDescriptionMap: Map<String, FieldDescriptionDto>? = null

    constructor(data: T?) : this(data, null)

    init {
        this.data = data
        this.fieldDescriptionMap = fieldDescriptionDtoMap
    }
}
